File: TransCoder_tokenized_test_set_functions/java/SUBARRAYS_DISTINCT_ELEMENTS.java
public static int sumoflength ( int [ ] arr, int n ) {
  Set < Integer > s = new HashSet < > ( );
  int j = 0, ans = 0;
  for ( int i = 0;
  i < n;
  i ++ ) {
    while ( j < n && ! s . contains ( arr [ j ] ) ) {
      s . add ( arr [ i ] );
      j ++;
    }
    ans += ( ( j - i ) * ( j - i + 1 ) ) / 2;
    s . remove ( arr [ i ] );
  }
  return ans;
}

File: TransCoder_tokenized_test_set_functions/cpp/SUBARRAYS_DISTINCT_ELEMENTS.cpp
int sumoflength ( int arr [ ], int n ) {
  unordered_set < int > s;
  int j = 0, ans = 0;
  for ( int i = 0;
  i < n;
  i ++ ) {
    while ( j < n && s . find ( arr [ j ] ) == s . end ( ) ) {
      s . insert ( arr [ j ] );
      j ++;
    }
    ans += ( ( j - i ) * ( j - i + 1 ) ) / 2;
    s . erase ( arr [ i ] );
  }
  return ans;
}